pkg/storage/streams/store.go

```diff
@@ -352,7 +377,27 @@ func (s *streamStore) List(ctx context.Context, prefix, startAfter, endBefore pa
 		metaFlags |= meta.UserDefined
 	}
 
-	segments, more, err := s.segments.List(ctx, prefix.Prepend("l"), startAfter, endBefore, recursive, limit, metaFlags)
+	encPrefix, err := prefix.EncryptWithBucket(s.rootKey)
+	if err != nil {
+		return nil, false, err
+	}
+
+	prefixNoBucket := paths.Path([]string(prefix)[1:])
```

Comment:
When deriving the key we should include the bucket part, as explained for the `EncryptWithBucket` method.